analyses/org.eclipse.tracecompass.incubator.executioncomparision.ui/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/ui/DifferentialFlameGraphView.java

```diff
@@ -0,0 +1,1261 @@
+import org.eclipse.tracecompass.tmf.core.signal.TmfSignalHandler;
+import org.eclipse.tracecompass.tmf.core.signal.TmfSignalManager;
+import org.eclipse.tracecompass.tmf.core.signal.TmfTraceClosedSignal;
+import org.eclipse.tracecompass.tmf.core.signal.TmfTraceSelectedSignal;
+import org.eclipse.tracecompass.tmf.core.timestamp.ITmfTimestamp;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+import org.eclipse.tracecompass.tmf.core.trace.TmfTraceManager;
+import org.eclipse.tracecompass.tmf.core.trace.TmfTraceUtils;
+import org.eclipse.tracecompass.tmf.ui.TmfUiRefreshHandler;
+import org.eclipse.tracecompass.tmf.ui.editors.ITmfTraceEditor;
+import org.eclipse.tracecompass.tmf.ui.symbols.TmfSymbolProviderUpdatedSignal;
+import org.eclipse.tracecompass.tmf.ui.views.SaveImageUtil;
+import org.eclipse.tracecompass.tmf.ui.views.TmfView;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.TimeGraphViewer;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.ITimeEvent;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.ITimeGraphEntry;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.NamedTimeEvent;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.NullTimeEvent;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.TimeEvent;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.TimeGraphEntry;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.model.TimeGraphEntry.Sampling;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.widgets.TimeGraphControl;
+import org.eclipse.tracecompass.tmf.ui.widgets.timegraph.widgets.Utils.TimeFormat;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchPartSite;
+import org.eclipse.ui.progress.IWorkbenchSiteProgressService;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+
+/**
+ * The differential flame graph used in the execution comparison view. It is
+ * based on the flame graph view and is used to compare two different
+ * executions.
+ *
+ * @author Fateme Faraji Daneshgar
+ *
+ */
+@SuppressWarnings("restriction")
+public class DifferentialFlameGraphView extends TmfView {
+
+    /**
+     * ID of the view
+     */
+    public static final String ID = DifferentialFlameGraphView.class.getPackage().getName() + ".diffflamegraphView"; //$NON-NLS-1$
+    private static final int DEFAULT_BUFFER_SIZE = 3;
+    private static final String DIRTY_UNDERFLOW = "Dirty underflow"; //$NON-NLS-1$
+    /**
+     * the Logger that is used in multipleDensityView class
+     */
+    protected static final Logger LOGGER = Logger.getLogger(DifferentialFlameGraphView.class.getName());
+
+    private @Nullable DifferentialWeightedTreeProvider<?> fDataProvider;
+    private @Nullable ITimeGraphDataProvider<TimeGraphEntryModel> fdataProviderGroup = null;
```

Comment:
Should be FlameChartEntryModel instead of TimeGraphEntryModel

Reply:
Done 👍